src/lib/crypto/kimchi_backend/gadgets/common.ml

```diff
@@ -0,0 +1,148 @@
+(* Common gadget helpers *)
+
+open Core_kernel
+
+(* field_bits_to_field - Create a field element from contiguous bits of another
+ *
+ *   Inputs:
+ *     field_element: source field element
+ *     start:         zero-indexed starting bit offset (or -1 to denote the last bit)
+ *     stop:          zero-indexed stopping bit index (or -1 to denote the last bit)
+ *
+ *   Output:
+ *     New field element created from bits [start, stop) of field_element input,
+ *     placed into the lowest possible bit position, like so
+ *
+ *        start     stop
+ *             \   /
+ *       [......xxx.....] field_element
+ *       [...........xxx] output
+ *       msb          lsb
+ *)
+let field_bits_to_field (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    (field_element : Circuit.Field.t) (start : int) (stop : int) =
+  let open Circuit in
+  (* Check range is valid *)
+  if stop <> -1 && stop <= start then
+    invalid_arg "stop offset must be greater than start offset" ;
```

Comment:
One could also go with
```suggestion
  assert (stop <> -1 && stop <= start);
```

The one difference is that we compile without assertions.

Reply:
OK, then I will leave it as `invalid_arg`